### PR TITLE
Hide installed packages from suggestions when package is not found

### PR DESCRIPTION
**Pull request by xy2z**
This solves #8280

### CODE REVIEW

File: src/Composer/Command/InitCommand.php [hunk 1]
```diff
@@ -792,7 +792,13 @@ private function findSimilar($package)
         }
         $similarPackages = array();
 
+        $installedRepo = $this->getComposer()->getRepositoryManager()->getLocalRepository();
+
         foreach ($results as $result) {
+            if ($installedRepo->hasPackageName($result['name'])) {
```
Comment (Seldaek) [inline src/Composer/Command/InitCommand.php]: Pleasse rather use `->findPackage($result['name'], '*')` to avoid adding a new method to all repositories.

Comment (xy2z) [reply src/Composer/Command/InitCommand.php]: Fixed

**Comment by xy2z** [comment]
I'll look into it 👍

**Comment by xy2z** [comment]
@Seldaek This should be fixed now

**Comment by Seldaek** [comment]
Thanks, simplified the code a little so it only loops once instead of on every package search, but otherwise patch was pretty good

**Pull request merged**